src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -414,7 +414,7 @@ ExpressionSyntax CreateExpressionToInline()
 
                     return SyntaxFactory.ArrayCreationExpression(arrayType, arrayInitializer);
                 }
-                else if (isVar && expression is ObjectCreationExpressionSyntax or ArrayCreationExpressionSyntax or CastExpressionSyntax)
+                else if (isVar && expression is ObjectCreationExpressionSyntax or ArrayCreationExpressionSyntax or CastExpressionSyntax or InvocationExpressionSyntax)
```

Comment:
That is most likely not the right fix unfortunatelly

Reply:
so what you suggest?

Reply:
i agree with DoctorKrolic.  thsi cahnge doesn't really make sense.  the comment below this explains the rational for why we don't need to add the cast.  however, for an invocatino expression we still *do* need to cast to preserve semantics in many cases.

Reply:
@CyrusNajmabadi  can i have an example when we need to do the cast on InvocationExpressionSyntax ?

Reply:
Ah, i think i missed that this iwas the `is var` case.  In that case, it's likely safe to whitelist invocations as well.  Can you update the comment inside to make this clearer?  Thanks!

Reply:
done!